controllers/argocd_metrics_controller.go

```diff
@@ -292,6 +308,94 @@ func (r *ArgoCDMetricsReconciler) createPrometheusRuleIfAbsent(namespace string,
 	return err
 }
 
+func (r *ArgoCDMetricsReconciler) reconcileDashboards(reqLogger logr.Logger) error {
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: dashboardNamespace}, &corev1.Namespace{})
+	if err != nil {
+		reqLogger.Info("Monitoring dashboards are not supported on this cluster, skipping dashboard installation",
+			"Namespace", dashboardNamespace)
+		return nil
+	}
+
+	entries, err := dashboards.ReadDir(dashboardFolder)
+	if err != nil {
+		reqLogger.Error(err, "Could not read list of embedded dashboards")
+		return err
+	}
+
+	for _, entry := range entries {
+		reqLogger.Info("Processing dashboard", "Namespace", dashboardNamespace, "Name", entry.Name())
+
+		if !entry.IsDir() {
+			dashboard, err := newDashboardConfigMap(entry.Name(), dashboardNamespace)
+			if err != nil {
+				reqLogger.Info("There was an error creating dashboard ", "Namespace", dashboardNamespace, "Name", entry.Name())
+				return err
+			}
+
+			existingDashboard := &corev1.ConfigMap{}
+
+			err = r.Client.Get(context.TODO(), types.NamespacedName{Name: dashboard.Name, Namespace: dashboardNamespace}, existingDashboard)
+			if err == nil {
+				reqLogger.Info("A dashboard instance already exists",
+					"Namespace", existingDashboard.Namespace, "Name", existingDashboard.Name)
+
+				// See if we need to reconcile based on dashboard data only to allow users
+				// to disable dashboard via label if so desired. Note that disabling it
+				// will be reset if dashboard changes in newer version of operator.
+				if existingDashboard.Data[entry.Name()] != dashboard.Data[entry.Name()] {
+					reqLogger.Info("Dashboard data does not match expectation, reconciling",
+						"Namespace", dashboard.Namespace, "Name", dashboard.Name)
+					err := r.Client.Update(context.TODO(), dashboard)
+					if err != nil {
+						reqLogger.Error(err, "Error updating dashboard",
+							"Namespace", dashboard.Namespace, "Name", dashboard.Name)
+						return err
+					}
+				}
+				continue
+			}
+
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating new dashboard",
+					"Namespace", dashboard.Namespace, "Name", dashboard.Name)
+				err := r.Client.Create(context.TODO(), dashboard)
+				if err != nil {
+					reqLogger.Error(err, "Error creating a new dashboard",
+						"Namespace", dashboard.Namespace, "Name", dashboard.Name)
+					// Should we return an error or continue with other dashboards?
+					return err
```

Comment:
I think we should continue with other dashboards here, unless they only make sense in context of each other 
If that is not that case and there is value in each dashboard independently, we should not return error here

same for line 332 as well